### PR TITLE
chore(ci): smoother deno_std release

### LINKED ISSUE #2078: Improvements to deno_std release
1. Similar to what we have in the CLI, we should have a workflow that automatically updates the `Releases.md` and bumps versions.ts. This should open a PR and have a checkbox task list that asks if they tested std with the current CLI.
2. On merge, we should have the release automatically created with the new contents from `Releases.md`

### PR DESCRIPTION
**Pull request by dsherret**
1. This adds a new version_bump.yml workflow similar to what is found in the CLI repo.
   1. On workflow run, this will bump the version in version.ts and update Releases.md
   2. Then it will automatically open a PR for us to review and follow the checklist.
2. After on PR merge to main, a draft PR release will automatically be created for us to review and then publish.

Closes #2078

### CODE REVIEW

File: _tools/release/02_create_pr.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env -S deno run --allow-read --allow-write --allow-env --allow-net --allow-run=git --no-check
```
Comment (dsherret) [inline _tools/release/02_create_pr.ts:1]: Untested, but copied from the CLI. Please review carefully.

File: _tools/release/03_release.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env -S deno run --allow-read --allow-write --allow-env --allow-net --allow-run=git --no-check
```
Comment (dsherret) [inline _tools/release/03_release.ts:1]: Untested, but adapted from elsewhere. Please review carefully.

File: _tools/release/01_bump_version.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env -S deno run --allow-read --allow-write --allow-run=git --no-check
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+import { getReleasesMdFile, loadRepo, VersionFile } from "./repo.ts";
+
+const repo = await loadRepo();
+const versionFile = new VersionFile();
+const currentVersion = versionFile.version;
+const newVersion = versionFile.version.inc(getVersionIncrementKind());
```
Comment (bartlomieju) [inline _tools/release/01_bump_version.ts:8]: Just to ensure, this is being incremented without using `v` prefix?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: _tools/release/01_bump_version.ts (hunk 3 above)
Comment (dsherret) [reply _tools/release/01_bump_version.ts:8]: Yup and this will tag without a `v` prefix.

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by kt3k** [comment]
I wonder if using actions like [this](https://github.com/peter-evans/create-pull-request) make sense

**Comment by dsherret** [comment]
@kt3k I had initially tried that on my CLI PR doing this, but it's much easier to just keep everything in the Deno script because then nothing needs to be passed around between steps. Passing around strings between GH action steps is a pain and passing around multi-line strings is a nightmare.

**Comment by kt3k** [comment]
@dsherret
> Passing around strings between GH action steps is a pain and passing around multi-line strings is a nightmare.

I see. That'd be more complex than this version. Thanks for the clarification!

**Pull request merged**